src/lib/y2network/sysconfig/interface_file.rb

```diff
@@ -200,35 +246,24 @@ def path
         SYSCONFIG_NETWORK_PATH.join("ifcfg-#{interface}")
       end
 
-      # Returns the IP address if defined
-      #
-      # @return [IPAddr,nil] IP address or nil if it is not defined
-      def ip_address
-        str = fetch("IPADDR")
-        str.nil? || str.empty? ? nil : IPAddress.new(str)
-      end
-      alias_method :ip_address, :ipaddr
-
-      # Fetches a key
+      # Loads values from the configuration file
       #
-      # @param key [String] Interface key
-      # @return [Object] Value for the given key
-      def fetch(key)
-        path = Yast::Path.new(".network.value.\"#{interface}\".#{key}")
-        Yast::SCR.Read(path)
+      # @return [Hash<String, Object>] All values from the file
+      def load
```

Comment:
So the load method loads all the attributes even if it does not belongs to an specific type isn't it?. Thus, inferring the type from the current values is harder as there is no distinction between present or not.

Reply:
Can be inferred from `@defined_variables so forget what I said :)